# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/SharedSubResourceCache.cpp
# Commit: 313c1f782833
# Full Hash: 313c1f7828336b6163271454aaface96f0e3fc96
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-10-02 09:50:09
# Regressor Bug: 1899734
# File Overlap Count: 1
# Description:
#   Bug 1899734 - Part 5: Make SharedSubResourceCache capable of handling CacheablePerformanceTimingData. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D223351
# ==============================================================================

diff -r baa464d5f255 -r 313c1f782833 layout/style/SharedSubResourceCache.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/layout/style/SharedSubResourceCache.cpp	Tue Oct 01 12:17:22 2024 +0000
@@ -0,0 +1,26 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "SharedSubResourceCache.h"
+
+#include "mozilla/dom/CacheablePerformanceTimingData.h"
+#include "nsCOMPtr.h"
+#include "nsIHttpChannel.h"
+#include "nsIRequest.h"
+#include "nsITimedChannel.h"
+
+namespace mozilla {
+
+SubResourceNetworkMetadataHolder::SubResourceNetworkMetadataHolder(
+    nsIRequest* aRequest) {
+  nsCOMPtr<nsITimedChannel> timedChannel = do_QueryInterface(aRequest);
+  if (timedChannel) {
+    nsCOMPtr<nsIHttpChannel> httpChannel = do_QueryInterface(aRequest);
+    mPerfData.emplace(timedChannel, httpChannel);
+  }
+}
+
+}  // namespace mozilla